ui/app/adapters/pki/pki-role-engine.js

```diff
@@ -1,3 +1,24 @@
 import PkiRoleAdapter from './pki-role';
+import { encodePath } from 'vault/utils/path-encoding-helpers';
 
-export default class PkiRoleEngineAdapter extends PkiRoleAdapter {}
+export default class PkiRoleEngineAdapter extends PkiRoleAdapter {
+  _urlForRole(backend, id) {
```

Comment:
handling the createRecord method here instead of following the old pattern in the pki-role adapter because it has a different snapshot due to its use of lazyPaginatedQuery. It's a rabbit hole, suffice it to say, I needed to make my own, which is nice because I can use a little more update logic.

Reply:
That's a good idea, however in the end I want to remove all of the previous pki adapters. Include the PKI role adapter. Which brings up a good point. I'm going to not extend the PkiRole adapter here, but instead extend the application adapter. I'll go ahead and make that change.

Reply:
Ok ya it's probably best to copy the entire `PkiRoleAdapter` and change what you need then.